fe/src/main/java/org/apache/doris/planner/SingleNodePlanner.java

```diff
@@ -1328,6 +1362,12 @@ private PlanNode createScanNode(Analyzer analyzer, TableRef tblRef)
         analyzer.materializeSlots(scanNode.getConjuncts());
 
         scanNodes.add(scanNode);
+        List<ScanNode> scanNodeList = selectStmtToScanNodes.get(selectStmt);
+        if (scanNodeList == null) {
+            scanNodeList = Lists.newArrayList();
+        }
+        scanNodeList.add(scanNode);
+        selectStmtToScanNodes.put(selectStmt, scanNodeList);
```

Comment:
move `selectStmtToScanNodes.put(selectStmt, scanNodeList);` to the above `if` clause.